docs/guide/api-hmr.md

```diff
@@ -30,7 +30,7 @@ interface ViteHotContext {
 
   dispose(cb: (data: any) => void): void
   decline(): void
-  invalidate(): void
+  invalidate(reason?: string): void
```

Comment:
I like this 👍🏼
I think we should pass the reason as `options: { reason?: string }` here though, just in case we need to extend the API in the future.

Reply:
We are missing updating the `reason` param here to options.

Reply:
Oups MB. I will update

Reply:
Sorry Arnaud, after discussing with Evan, looks like your original version is preferred. We don't know if we will ever extend this method, and if we do, we could overload to take both a string or object. Or maybe even follow kind of what Error does `later new Error(message, options)`. It is useful for debugging to always have a reason or message so not that bad to be pushing people to add that string.

So, let's revert to your original version passing a single string. But we have green light to merge this after you do 👍🏼

Reply:
What about logging invalidate even when there is no message?

Should I call the parameter message or reason?

Reply:
Maybe `message` is more aligned with other similar APIs? `reason` sounds more like `cause` https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Error/cause, which has other implications.
But no strong opinions here.

About logging when invalidating, I think it makes sense, yes. It would be interesting to see if there is useful info we could always show. Let's review if nobody complains because it is too noisy. But seeing the message could be good to push more folks in the ecosystem to use the new `message` param.